### PR TITLE
Add batch operations

### PR DESCRIPTION
**Pull request by AMHOL**
#### Example:

```ruby
class User < ActiveRecord::Base
  include Nazrin::Searchable

  searchable_configure do |config|
    # My config
  end

  searchable do
    fields [:name, :email]
  end
end
# Create some users
user_1 = User.create(name: 'Jane', email: 'no-reply@jane-fake-email.com')
user_2 = User.create(name: 'Joe', email: 'no-reply@joe-fake-email.com')
user_3 = User.create(name: 'John', email: 'no-reply@john-fake-email.com')
# Delete a user
user_3.destroy

User.nazrin_batch_operation(
  add: [user_1, user_2],
  delete: [user_3]
)
```

### CODE REVIEW

File: lib/nazrin/searchable.rb [hunk 1]
```diff
@@ -81,6 +81,22 @@ def nazrin_add_document(obj)
           obj.send(:id), nazrin_eval_field_data(obj))
       end
 
+      def nazrin_batch_operation(type_objects_mapping)
+        operations = type_objects_mapping.each_with_object({}) do |(type, objects), hash|
+          if type.to_sym == :add
+            hash[:add] = objects.map do |obj|
+              [obj.send(:id), nazrin_eval_field_data(obj)]
+            end
+          else
```
Comment (tsuwatch) [inline lib/nazrin/searchable.rb]: I would like you to specify `delete` for typo

File: lib/nazrin/searchable.rb [hunk 2]
```diff
@@ -81,6 +81,22 @@ def nazrin_add_document(obj)
           obj.send(:id), nazrin_eval_field_data(obj))
       end
 
+      def nazrin_batch_operation(type_objects_mapping)
```
Comment (tsuwatch) [inline lib/nazrin/searchable.rb]: I want alias method like https://github.com/AMHOL/nazrin/blob/1575066637d1c8fd03b336ee307978c49e790bc5/lib/nazrin/searchable.rb#L32

**Comment by AMHOL** [comment]
Thanks for review @tsuwatch, I also added `create` as an alias of `add` and `destroy` as an alias of `delete`, and made it raise an error if the operations are not one of the above.

**Comment by AMHOL** [comment]
Let me know if you want me to squash commits

**Comment by tsuwatch** [comment]
Why need `create` and `destroy`?

**Comment by AMHOL** [comment]
I just thought I may as well add them as aliases as they're familiar methods for people using Rails, I can remove if you like?

File: lib/nazrin/document_client.rb [hunk 3]
```diff
@@ -37,5 +39,40 @@ def delete_document(id)
...
+        when :add, :create
+          tuple.each do |id, field_data|
+            arr.push(
+              type: 'add',
+              id: id,
+              fields: field_data
+            )
+          end
+        when :delete, :destroy
+          tuple.map do |id|
```
Comment (AMHOL) [inline lib/nazrin/document_client.rb]: Gonna change this and the other two to `each` rather than `map` as `map` implies that the return value matters.

**Comment by tsuwatch** [comment]
Hmm, I think It is not necessary to support aliases.

**Comment by tsuwatch** [comment]
Nice work, thank you!!!

**Comment by AMHOL** [comment]
Thanks @tsuwatch :smile:

**Pull request merged**